TF/src/common/Graph.py

```diff
@@ -0,0 +1,137 @@
+
+from queue import Queue
+
+# Graph Module to store adjacency list and graph related attributed
+# Nodes will be a list of operators, edges a list of tensors
+# start_nodes is a list of indices referencing nodes and edges, denoting entry nodes
+# adj_list is a dictionary of the form {src node : list of ([edge, dest_node])} 
+# where src_node, edge and dest_node and indices referencing nodes and edges
+class Graph:
+    def __init__(self, nodes, start_node_indices, edges, adj_list, model_name, category):
+        self.nodes = nodes
+        self.start_node_indices = start_node_indices
+        self.edges = edges
+        self.adj_list = adj_list
+        self.category = category
+        self.model_name = model_name
+        self.num_inputs = len(self.start_node_indices)
+        self.num_outputs = self.calc_num_outputs()
+        self.max_fan_in = self.calc_max_fan_in()
+        self.max_fan_out = self.calc_max_fan_out()
+
+    def calc_num_outputs(self):
+        ret = 0
+
+        # Number of proxy nodes for output
+        for node in self.nodes:
+            if node.label == 'Output_Placeholder':
+                ret += 1
+
+        return ret
+
+    def calc_max_fan_out(self):
+        ret = 0
+
+        for src_node_index in range(len(self.nodes)):
+            if src_node_index in self.adj_list:
+                ret = max(ret, len(self.adj_list[src_node_index]))
+
+        return ret
+
+    def calc_max_fan_in(self):
+        in_degree = [0] * len(self.nodes)
+        ret = 0
+
+        for src_node_index in range(len(self.nodes)):
+            if src_node_index in self.adj_list:
+                for (_, dest_node_index) in self.adj_list[src_node_index]:
+                    in_degree[dest_node_index] += 1
+                    ret = max(ret, in_degree[dest_node_index])
+ 
+        return ret    
+
+    # Print graph in BFS order
+    def print_graph(self):
```

Comment:
This is pretty much same as in TFLite/src/common/Graph.py. How about having a common util where we can move such common utility and reduce code duplication?

Reply:
Fixed in commit with message 'Restructure code', changed file structure for common files.